build_system/src/test.rs

```diff
@@ -1,15 +1,1176 @@
+            "mir/mir_heavy/issue-miri-1910.rs",
+        ]
+        .iter()
+        .any(|to_ignore| path_str.ends_with(to_ignore))
+}
+
+fn test_rustc_inner<F>(env: &Env, args: &TestArg, callback: F) -> Result<(), String>
+where
+    F: Fn() -> Result<bool, String>,
+{
+    // FIXME: create a function "display_if_not_quiet" or something along the line.
+    println!("[TEST] rust-lang/rust");
+    let mut env = env.clone();
+    setup_rustc(&mut env, args)?;
+
+    walk_dir(
+        "rust/tests/ui",
+        |dir| {
+            let dir_name = dir.file_name().and_then(|name| name.to_str()).unwrap_or("");
+            if [
+                "abi",
+                "extern",
+                "unsized-locals",
+                "proc-macro",
+                "threads-sendsync",
+                "borrowck",
+                "test-attrs",
+            ]
+            .iter()
+            .any(|name| *name == dir_name)
+            {
+                std::fs::remove_dir_all(dir).map_err(|error| {
+                    format!("Failed to remove folder `{}`: {:?}", dir.display(), error)
+                })?;
+            }
+            Ok(())
+        },
+        |_| Ok(()),
+    )?;
+
+    fn dir_handling(dir: &Path) -> Result<(), String> {
+        walk_dir(dir, dir_handling, file_handling)
+    }
+    fn file_handling(file: &Path) -> Result<(), String> {
+        let path_str = file.display().to_string().replace("\\", "/");
+        if !path_str.ends_with(".rs") {
+            return Ok(());
+        } else if should_not_remove_test(&path_str) {
+            return Ok(());
+        } else if should_remove_test(file, &path_str) {
+            return std::fs::remove_file(file)
+                .map_err(|error| format!("Failed to remove `{}`: {:?}", file.display(), error));
+        }
+        let file_content = std::fs::read_to_string(file)
+            .map_err(|error| format!("Failed to read `{}`: {:?}", file.display(), error))?;
+        if should_remove_ui_test(&file_content) {
+            std::fs::remove_file(file)
+                .map_err(|error| format!("Failed to remove `{}`: {:?}", file.display(), error))?;
+        }
+        Ok(())
+    }
+
+    let rust_path = Path::new("rust");
+
+    walk_dir(rust_path.join("tests/ui"), dir_handling, file_handling)?;
+    let file = rust_path.join("tests/ui/consts/const_cmp_type_id.rs");
+    std::fs::remove_file(&file)
+        .map_err(|error| format!("Failed to remove `{}`: {:?}", file.display(), error))?;
+    let file = rust_path.join("tests/ui/consts/issue-73976-monomorphic.rs");
+    std::fs::remove_file(&file)
+        .map_err(|error| format!("Failed to remove `{}`: {:?}", file.display(), error))?;
+
+    if !callback()? {
+        // FIXME: create a function "display_if_not_quiet" or something along the line.
+        println!("Keeping all UI tests");
+    }
+
+    let nb_parts = args.nb_parts.unwrap_or(0);
+    if nb_parts > 0 {
+        let current_part = args.current_part.unwrap_or(0);
```

Comment:
It might be more useful to do `expect()` instead of `unwrap_or(0)` so we would want to know if we forget to specify the part.